google/gax/__init__.py

```diff
@@ -673,3 +670,5 @@ def _execute_clbks(self):
 
         for done_clbk in self._done_clbks:
             _try_callback(self, done_clbk)
+
+        self._done_clbks = []
```

Comment:
I think it's bad practice (and also causes bad things to happen) to update a list that is being iterated on in a for loop. This would be better if it was a queue that was just getting shifted for each callback which would both empty the list after all callbacks were called and would avoid the updating a list while in a for loop.

In _OperationFuture constructor:
```py
_done_clbks = deque()
```

In this function:
```py
while _done_clbks:
    done_clbk = _done_clbks.popleft()
    _try_callback(self, done_clbk)